### ISSUE TITLE
Record<> ordering with null field

### ISSUE THREAD

**Issue by gwintering** [louthy/language-ext#491]
Records don't provide the expected ordering when a field in both instances is `null`:
```csharp
class Disorder : Record<Disorder>
{
     public string Present = "Here";
     public string Absent = null;
}

var a = new Disorder();
var b = new Disorder();

a == b // true -- Expected true
a.GetHashCode() == b.GetHashCode() // true -- Expected true
a.CompareTo(b) // 1 -- Expected 0
```

I know that `null` in a record is not the purist thing, but until C# gives us a way to banish `null` it will remain common.

For comparison, tuples **do** provide the expected ordering:
```csharp
var a = ("Here", null as string);
var b = ("Here", null as string);

a == b // true
a.GetHashCode() == b.GetHashCode() // true
a.CompareTo(b) // true
```

**Comment by louthy** [comment]
Indeed, that looks wrong.  I’ll check

**Comment by louthy** [comment]
This is fixed in `v3.0.29`, along with a general performance improvement for .NET Framework users. If we ever get `CompileToMethod` in .NETStandard then that improvement can be brought over.

**Comment by gwintering** [comment]
Wow, thanks for the speedy fix Paul!
